django/forms/widgets.py

```diff
@@ -787,6 +787,19 @@ class CheckboxSelectMultiple(RendererMixin, SelectMultiple):
     renderer = CheckboxFieldRenderer
     _empty_value = []
 
+    def build_attrs(self, extra_attrs=None, **kwargs):
+        attrs = super(CheckboxSelectMultiple, self).build_attrs(extra_attrs, **kwargs)
+        # The required attribute comes from the bound field if the form field
+        # is required. When required is set on multiple checkboxes with the
+        # same name, the browser will require each checkbox be checked to
+        # submit the form. This is not the desired validation, so remove the
+        # attribute.
+        try:
+            del attrs['required']
+        except KeyError:
+            pass
```

Comment:
In these cases, I tend to follow the suggested Python style [easier to ask for forgiveness than permission](https://docs.python.org/3/glossary.html#term-eafp). But if the Django project prefers a different style, I have no problem changing it.

Given this, would you still prefer I change this?